resources/ts/filters/filter-dropdown.ts

```diff
@@ -1,51 +1,100 @@
 import submitForm from '../form/form-handler';
+import isReducedMotion from '../common/reduced-motion-detection';
 
 const DROPDOWN_SELECTOR = '.fau-dropdown';
 const DROPDOWN_TOGGLE_SELECTOR = '.fau-dropdown__toggle';
 const DROPDOWN_CONTENT_SELECTOR = '.fau-dropdown__content';
 const CLICKAWAY_WINDOW_WIDTH_THRESHOLD = 768; // close on click away only works on screen sizes above this value
 
-const closeDropDown = ( dropdown: HTMLElement ) => {
-	dropdown.setAttribute( 'aria-expanded', 'false' );
-	submitForm();
-};
+class Dropdown {
+	private dropdown: HTMLElement;
+	private toggle: HTMLElement;
+	private content: HTMLElement;
+	private checkboxes: HTMLInputElement[];
+	private initialState: boolean[] = [];
 
-const toggleDropdown = ( dropdown: HTMLElement ) => {
-	const isAriaExpanded = dropdown.getAttribute( 'aria-expanded' ) === 'true';
+	constructor( dropdown: HTMLElement ) {
+		this.dropdown = dropdown;
+		this.toggle = dropdown.querySelector( DROPDOWN_TOGGLE_SELECTOR )!;
+		this.content = dropdown.querySelector( DROPDOWN_CONTENT_SELECTOR )!;
+		this.checkboxes = Array.from(
+			this.content.querySelectorAll( 'input[type="checkbox"]' )
+		);
 
-	dropdown.setAttribute( 'aria-expanded', isAriaExpanded ? 'false' : 'true' );
+		this.init();
+	}
 
-	if ( isAriaExpanded ) {
-		submitForm();
+	private recordInitialState() {
+		this.initialState = this.checkboxes.map(
+			( checkbox ) => checkbox.checked
+		);
 	}
-};
 
-const registerClickListeners = () => {
+	private hasStateChanged(): boolean {
+		const currentState = this.checkboxes.map(
+			( checkbox ) => checkbox.checked
+		);
+
+		return ! this.initialState.every(
+			( value, index ) => value === currentState[ index ]
+		);
```

Comment:
```suggestion
		return this.initialState.some(
			( value, index ) => value !== currentState[ index ]
		);
```
I would use `some` instead.